sdk/eventhub/azure-eventhub/azure/eventhub/_transport/_pyamqp_transport.py

```diff
@@ -542,7 +563,11 @@ def _create_eventhub_exception(exception):
         elif isinstance(exception, TimeoutError):
             error = ConnectionLostError(str(exception), exception)
         else:
-            error = EventHubError(str(exception), exception)
+            if isinstance(exception, FileNotFoundError) and exception.filename2 == "consumer":
```

Comment:
Rather than put this attribute on the error - can we not simply pass an optional `is_consumer` flag into the method signature?

Reply:
updated + adding a comment for posterity